code-challenges.test.js

```diff
@@ -51,6 +80,32 @@ const studyMinutesWeek2 = {
 }
 // Expected output: [10, 15, 20, 45, 60, 65, 100]
 
+describe("studyTime", () => {
+  // studyTime is the string argument for the describe function
+  it("takes in an object and returns an array of the object's values sorted from least to greatest", () => {
+  // in the it statement, the objects array's are sorted from least to greatest
+    expect(studyTime(studyMinutesWeek1)).toEqual([15, 15, 20, 30, 30, 60, 90])
+    expect(studyTime(studyMinutesWeek2)).toEqual([10, 15, 20, 45, 60, 65, 100])
+  // the two expect statements are calling the function for each object's and making comparisons to the output
+  })
+}) // FAIL:  ReferenceError: studyTime is not defined
+
 // b) Create the function that makes the test pass.
 
-// Pseudo code:
+const studyTime = (studyMin) => {
+  // function studyTime uses parameter studyMin for the argument
+  const studyMinutesArray = Object.values(studyMin);
+  // studyMinutesArray contains the value of Object.values(studyMin), which extracts the values of the keys in the object
+  studyMinutesArray.sort((a, b) => a - b);
+  // .sort() arranges the elements in order. however, when using just .sort(), the array isn't arrange in numeric order. (a, b) => a - b) is used to compare the elements with the array and states whether the element is bigger or smaller and arranges them accordingly.
+  return studyMinutesArray;
+} // PASS: ✓ takes in an object and returns an array of the object's values sorted from least to greatest (2 ms)
+
+// const result1 = studyTime(studyMinutesWeek1);
+// console.log(result1); // [15, 15, 20, 30, 30, 60, 90]
+
+// const result2 = studyTime(studyMinutesWeek2);
+// console.log(result2); // [10, 15, 20, 45, 60, 65, 100]
+
+
+// Pseudo code: I used Microsoft Bing AI to assist with solving the problem.
```

Comment:
Great! AI is a wonderful tool to be able to code faster. However it still requires an understanding of the code.